internal/hierarchyconfig/validator.go

```diff
@@ -129,6 +132,13 @@ func (v *Validator) handle(ctx context.Context, log logr.Logger, req *request) a
 		return webhooks.Deny(metav1.StatusReasonForbidden, reason)
 	}
 
+	allErrs := validateManagedMeta(req.hc)
+	if len(allErrs) > 0 {
+		gk := schema.GroupKind{Group: api.GroupVersion.Group, Kind: "HierarchyConfiguration"}
+		err := apierrors.NewInvalid(gk, req.hc.Name, allErrs)
+		return webhooks.DenyFromAPIError(err)
+	}
+
```

Comment:
Can we shorten this even more? I like the lower-level function to encapsulate as much information as possible so that the top-level function is easier to read, e.g:

```
if denied := validateManagedMeta(req.hc); denied != nil {
  return denied
}
```

Now the caller (`handle()`) doesn't have to know anything about the _type_ of information coming out of `validateManagedMeta` other than that it's ready to be returned to _its_ caller (`Handle()`).

Reply:
There reason I left the `allErrs` there, is to make it more obvious how to include more field errors in the same response. And to avoid more refactoring if/when more fields are added to `HierarchyConfiguration` that require validation in this webhook.

If we were using the high-level webhook framework provided by controller-runtime, `Error`/`field.ErrorList` is the "contract" between your webhook implementation and controller-runtime. That is another reason for keeping it as suggested IMO.

So here I do not agree with you, and would like to keep it as is. But if you insist..... 😉

Reply:
Interesting. Let's leave it as-is for now, but generally I prefer to keep the current state of the code as readable as possible, and save the flexibility until we actually use them in the PRs. But especially if you'll be adding that shortly, it's fine to go in like this.